obspy/core/stream.py

```diff
@@ -1797,6 +1822,38 @@ def select(self, network=None, station=None, location=None, channel=None,
         All other selection criteria that accept strings (network, station,
         location) may also contain Unix style wildcards (``*``, ``?``, ...).
         """
+        if inventory is not None:
+            trace_ids = []
+            start_dates = []
+            end_dates = []
+            for net in inventory.networks:
+                for sta in net.stations:
+                    for chan in sta.channels:
+                        id = '.'.join((net.code, sta.code,
+                                       chan.location_code, chan.code))
+                        trace_ids.append(id)
+                        start_dates.append(chan.start_date)
+                        end_dates.append(chan.end_date)
```

Comment:
So it will only keep Traces that have info in the inventory down to Channel level, it looks like? If thats the case, functionality is pretty similar to the lookup done in `Stream.remove_response()` which eventually calls `Inventory.get_channel_metadata(seed_id, datetime=None)` and it should likely be combined with that one to avoid code duplication

Reply:
Also `Inventory.get_contents()` has similar functionality, although missing start/end times of channels, but it might be possible to adapt that routine and avoid duplication.

Reply:
I tested several approaches and this one, which is similar to `Inventory.get_contents()`, is the fastest.

What should we do to avoid duplication? Is it fine to have an option for `Inventory.get_contents()` which returns also the start and end dates? 

Something like `Inventory.get_contents(channel_dates=False)`, which –if `True`– adds to the returned dictionary two more keys: `'channel_start_dates'` (with a list of start dates as value) and `'channel_end_dates'` (with a list of end dates as value).

What do you think?

Reply:
>What should we do to avoid duplication?

One thing I have been planning to look into (just haven't had time yet) is add a `get_stations` method to inventory with the same signature as [`fdsn.Client.get_stations`](https://docs.obspy.org/packages/autogen/obspy.clients.fdsn.client.Client.get_stations.html). Then all the filtering type methods can simply call `get_stations` with the appropriate keywords under the hood to avoid duplication. I have a functioning implementation [here](https://github.com/niosh-mining/obsplus/blob/master/obsplus/stations/get_stations.py) but it it doesn't yet support all the keywords. Looking closer at it, it doesn't even use pandas so it may be readily adaptable to this case if you want to use/improve it.

The idea of using the `get_x` as an *interface* for selecting data has proved very useful in obsplus. Its nice to write code which only needs to understand the type of data (waveform, station, event) and not the whether the data is a local or remote resource.

Reply:
In any case, whatever we do, it's gotta be done within the next couple days. If you cant get rid of duplications by then it stays as is for now. Wanna push the RC ASAP. :)

Reply:
"whatever we do" with "we" as in @claudiodsf  btw ;-P

Reply:
whelp.. its not exposed in a public helper routine.. so gotta get rid of duplications later i guess.. cant hold off RC for it now